src/Workspaces/Core/Desktop/Workspace/MSBuild/MSBuildProjectLoader.cs

```diff
@@ -259,8 +259,12 @@ private class LoadState
             private Dictionary<ProjectId, ProjectInfo> _projectIdToProjectInfoMap
                 = new Dictionary<ProjectId, ProjectInfo>();
 
-            private List<ProjectInfo> _projectInfoList
-                = new List<ProjectInfo>();
+            /// <summary>
+            /// Used to memoize results of <see cref="ProjectAlreadyReferencesProject"/> calls.
+            /// Reset any time internal state is changed.
+            /// </summary>
+            private Dictionary<ProjectId, Dictionary<ProjectId, bool>> ProjectAlreadyReferencesProjectMemo
```

Comment:
>rivate Dictionary<ProjectId, Dictionary<ProjectId, bool>> [](start = 13, length = 57)

Can this just be Dictionary<ProjectId, ProjectId> that corresponds to the inputs to the ProjectAlreadyReferencesProject method?